src/main/java/duke/utilityfunctions/Filereader.java

```diff
@@ -0,0 +1,31 @@
+package duke.utilityfunctions;
+
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.Todo;
+
+public class Filereader {
+    public Task generateTask(String input) {
+        Task task = null;
+        char taskIcon = input.charAt(1);
+        char taskStatus = input.charAt(4);
+        if(taskIcon == 'E'){
```

Comment:
[Also in some other parts of the code] Consider refactoring the magic literals into `final` variables.